openstack/networking/v2/extensions/bgpvpns/requests.go

```diff
@@ -0,0 +1,472 @@
+	Routes            []PortRoutes `json:"routes,omitempty"`
+	AdvertiseFixedIPs *bool        `json:"advertise_fixed_ips,omitempty"`
+}
+
+// ToPortAssociationCreateMap builds a request body from
+// CreatePortAssociationOpts.
+func (opts CreatePortAssociationOpts) ToPortAssociationCreateMap() (map[string]interface{}, error) {
+	return gophercloud.BuildRequestBody(opts, "port_association")
+}
+
+// CreatePortAssociation creates a new port association for a specified
+// BGP VPN.
+func CreatePortAssociation(ctx context.Context, client *gophercloud.ServiceClient, id string, opts CreatePortAssociationOptsBuilder) (r CreatePortAssociationResult) {
+	b, err := opts.ToPortAssociationCreateMap()
+	if err != nil {
+		r.Err = err
+		return
+	}
+	resp, err := client.Post(ctx, createPortAssociationURL(client, id), b, &r.Body, nil)
+	_, r.Header, r.Err = gophercloud.ParseResponse(resp, err)
+	return
+}
+
+// GetPortAssociation retrieves a specific port association by BGP VPN id
+// and port association id.
+func GetPortAssociation(ctx context.Context, c *gophercloud.ServiceClient, bgpVpnID string, id string) (r GetPortAssociationResult) {
+	resp, err := c.Get(ctx, getPortAssociationURL(c, bgpVpnID, id), &r.Body, nil)
+	_, r.Header, r.Err = gophercloud.ParseResponse(resp, err)
+	return
+}
+
+// DeletePortAssociation deletes a specific port association by BGP VPN id
+// and port association id.
+func DeletePortAssociation(ctx context.Context, c *gophercloud.ServiceClient, bgpVpnID string, id string) (r DeletePortAssociationResult) {
+	resp, err := c.Delete(ctx, deletePortAssociationURL(c, bgpVpnID, id), nil)
+	_, r.Header, r.Err = gophercloud.ParseResponse(resp, err)
+	return
+}
+
+// UpdatePortAssociationOptsBuilder allows extensions to add additional
+// parameters to the UpdatePortAssociation request.
+type UpdatePortAssociationOptsBuilder interface {
+	ToPortAssociationUpdateMap() (map[string]interface{}, error)
+}
+
+// UpdatePortAssociationOpts represents options used to update a BGP VPN
+// port association.
+type UpdatePortAssociationOpts struct {
+	Routes            *[]PortRoutes `json:"routes,omitempty"`
+	AdvertiseFixedIPs *bool         `json:"advertise_fixed_ips,omitempty"`
+}
+
+// ToPortAssociationUpdateMap builds a request body from
+// UpdatePortAssociationOpts.
+func (opts UpdatePortAssociationOpts) ToPortAssociationUpdateMap() (map[string]interface{}, error) {
+	return gophercloud.BuildRequestBody(opts, "port_association")
+}
+
+// UpdatePortAssociation updates a port association for a specified BGP VPN.
+func UpdatePortAssociation(ctx context.Context, client *gophercloud.ServiceClient, bgpVpnID string, id string, opts UpdatePortAssociationOptsBuilder) (r UpdatePortAssociationResult) {
+	b, err := opts.ToPortAssociationUpdateMap()
+	if err != nil {
+		r.Err = err
+		return
+	}
+	resp, err := client.Put(ctx, updatePortAssociationURL(client, bgpVpnID, id), b, &r.Body, &gophercloud.RequestOpts{
+		OkCodes: []int{200},
```

Comment:
Here too, [doc says the successful http code is 201](https://docs.openstack.org/api-ref/network/v2/#id857). Which is correct?

Reply:
Correct is the one, which is returned by API and passes functional tests :)